New Harvest-artifacts plugin; plus some minor changes on SitRep (color, typo, new screenshots)

MJ659600 opened the pull request.

modukon commented on content/artifacts/harvest-artifacts/plugin.js:
```diff
@@ -0,0 +1,337 @@
+      .filter((p) => p.planetLevel > myART.rarity)
+      .filter((p) => (df.getDist(srcId, p.locationId) < df.getMaxMoveDist(srcId, maxRangePct)))
+      .filter((p) => p.locationId !== srcId)
+      .sort((a, b) => { return df.getDist(srcId, a.locationId) - df.getDist(srcId, b.locationId) })
```
I see a lot of these .filter() chains
this could be rewritten to only use 1 .filter() and all conditions could be inside of it separated with a &&
I am not a js expert so please correct me if I am wrong, but i think every time the filter function creates a new array of planets and this requires more time.
Not saying that this needs to be changed, maybe there is no performance problem here.

MJ659600 replied:
wow. Difference is quite clear.  Thanks.  Learned something. This will be useful to know for bots.   For this code though, there is no performance issue.

MJ659600 commented:
Would love to know from experts if this is indeed true.   I am not a developer and new to js especially.  My code is largely adopted from other people's working codeSent from my Galaxy
-------- Original message --------From: modukon ***@***.***> Date: 8/24/21  2:32 AM  (GMT+08:00) To: darkforest-eth/plugins ***@***.***> Cc: MJ659600 ***@***.***>, Author ***@***.***> Subject: Re: [darkforest-eth/plugins] New Harvest-artifacts plugin; plus some minor changes on SitRep (color, typo, new screenshots) (#156) 
@modukon commented on this pull request.



In content/artifacts/harvest-artifacts/plugin.js:
> +  const myART = df.getArtifactWithId(artId);
+  if (!myART) { console.log(" --invalid ART"); return }
+
+  const source = df.getPlanetWithId(srcId);
+
+  if (source.owner != df.account) { console.log(" --Dont Own planet"); return }
+
+  //#1, try send to a RIP
+  if (tgtId === "") {
+    let targetList = df.getMyPlanets()
+      .filter((p) => df.getLocationOfPlanet(p.locationId))
+      .filter((p) => p.planetType == 3)  //only to RIP
+      .filter((p) => p.planetLevel > myART.rarity)
+      .filter((p) => (df.getDist(srcId, p.locationId) < df.getMaxMoveDist(srcId, maxRangePct)))
+      .filter((p) => p.locationId !== srcId)
+      .sort((a, b) => { return df.getDist(srcId, a.locationId) - df.getDist(srcId, b.locationId) })

I see a lot of these .filter() chains
this could be rewritten to only use 1 .filter() and all conditions could be inside of it separated with a &&
I am not a js expert so please correct me if I am wrong, but i think every time the filter function creates a new array of planets and this requires more time.
Not saying that this needs to be changed, maybe there is no performance problem here.

—You are receiving this because you authored the thread.Reply to this email directly, view it on GitHub, or unsubscribe.Triage notifications on the go with GitHub Mobile for iOS or Android.

Stx69 commented:
Hi MJ, nice I used your rawly previously code, and I think will be nice to have checkbox for function "deactivation of artefact." only this was worried me at past. I working on local game to be helpfull with trials.
 BR

Stx69 reviewed (changes requested):
For a harvest-artifacts really need be deactivation checkbox... We dont wanna lose artifacts.. :D

MJ659600 commented:
Removed references to SR_CFG and code related to "protecting from
harvesting".
Changed autoDeactivate default to "false".

If someone can give me some code samples to incorporate a checkbox I can
add it.  Don't laugh:  without a code example I'd take hours to figure out.

Also, I'd be happy if @thickforest or anyone else wants to incorporate the
ideas here and retire this code.










On Sun, Aug 29, 2021 at 5:04 AM Stx69 ***@***.***> wrote:

> ***@***.**** requested changes on this pull request.
>
> For a harvest-artifacts really need be deactivation checkbox... We dont
> wanna lose artifacts.. :D
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/darkforest-eth/plugins/pull/156#pullrequestreview-741056681>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ARO665AS6BDISORU5GSJ7PLT7FFOZANCNFSM5CU7W37Q>
> .
> Triage notifications on the go with GitHub Mobile for iOS
> <https://apps.apple.com/app/apple-store/id1477376905?ct=notification-email&mt=8&pt=524675>
> or Android
> <https://play.google.com/store/apps/details?id=com.github.android&referrer=utm_campaign%3Dnotification-email%26utm_medium%3Demail%26utm_source%3Dgithub>.
>
>

Stx69 commented:
> Removed references to SR_CFG and code related to "protecting from harvesting". Changed autoDeactivate default to "false". If someone can give me some code samples to incorporate a checkbox I can add it. Don't laugh: without a code example I'd take hours to figure out. Also, I'd be happy if @thickforest or anyone else wants to incorporate the ideas here and retire this code.
> […](#)
> On Sun, Aug 29, 2021 at 5:04 AM Stx69 ***@***.***> wrote: ***@***.**** requested changes on this pull request. For a harvest-artifacts really need be deactivation checkbox... We dont wanna lose artifacts.. :D — You are receiving this because you were mentioned. Reply to this email directly, view it on GitHub <[#156 (review)](https://github.com/darkforest-eth/plugins/pull/156#pullrequestreview-741056681)>, or unsubscribe <https://github.com/notifications/unsubscribe-auth/ARO665AS6BDISORU5GSJ7PLT7FFOZANCNFSM5CU7W37Q> . Triage notifications on the go with GitHub Mobile for iOS <https://apps.apple.com/app/apple-store/id1477376905?ct=notification-email&mt=8&pt=524675> or Android <https://play.google.com/store/apps/details?id=com.github.android&referrer=utm_campaign%3Dnotification-email%26utm_medium%3Demail%26utm_source%3Dgithub>.

You think for example so?

    let autoDeactivateCheckbox = document.createElement('input');
    autoDeactivateCheckbox.type = "Checkbox";
    autoDeactivateCheckbox.style.marginLeft = "10px";
    autoDeactivateCheckbox.checked = false;
    autoDeactivateCheckbox.id = "autoDeactivateCheckbox";
    this.autoDeactivateCheckbox = autoDeactivateCheckbox;

    container.appendChild(autoDeactivateCheckbox);

at function (autoDeactivateCheckbox)
autoDeactivateCheckbox.checked = true
!autoDeactivateCheckbox.checked = false

MJ659600 commented:
This is great.  Thanks! Got it working.
How do I add a label text on the same line of the checkbox?

On Mon, Aug 30, 2021 at 5:38 AM Stx69 ***@***.***> wrote:

> Removed references to SR_CFG and code related to "protecting from
> harvesting". Changed autoDeactivate default to "false". If someone can give
> me some code samples to incorporate a checkbox I can add it. Don't laugh:
> without a code example I'd take hours to figure out. Also, I'd be happy if
> @thickforest <https://github.com/thickforest> or anyone else wants to
> incorporate the ideas here and retire this code.
> … <#m_-203379824631747869_>
> On Sun, Aug 29, 2021 at 5:04 AM Stx69 *@*.*> wrote: @.** requested
> changes on this pull request. For a harvest-artifacts really need be
> deactivation checkbox... We dont wanna lose artifacts.. :D — You are
> receiving this because you were mentioned. Reply to this email directly,
> view it on GitHub <#156 (review)
> <https://github.com/darkforest-eth/plugins/pull/156#pullrequestreview-741056681>>,
> or unsubscribe
> https://github.com/notifications/unsubscribe-auth/ARO665AS6BDISORU5GSJ7PLT7FFOZANCNFSM5CU7W37Q
> . Triage notifications on the go with GitHub Mobile for iOS
> https://apps.apple.com/app/apple-store/id1477376905?ct=notification-email&mt=8&pt=524675
> or Android
> https://play.google.com/store/apps/details?id=com.github.android&referrer=utm_campaign%3Dnotification-email%26utm_medium%3Demail%26utm_source%3Dgithub
> .
>
> You think for example so?
>
> let autoDeactivateCheckbox = document.createElement('input');
>
> autoDeactivateCheckbox.type = "Checkbox";
>
> autoDeactivateCheckbox.style.marginLeft = "10px";
>
> autoDeactivateCheckbox.checked = false;
>
> autoDeactivateCheckbox.id = "autoDeactivateCheckbox";
>
> this.autoDeactivateCheckbox = autoDeactivateCheckbox;
>
>
>
> container.appendChild(autoDeactivateCheckbox);
>
>
> at function (autoDeactivateCheckbox)
> autoDeactivateCheckbox.checked = true
> !autoDeactivateCheckbox.checked = false
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/darkforest-eth/plugins/pull/156#issuecomment-907878396>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ARO665DT3Q3DGAXDAXMDKFLT7KSEXANCNFSM5CU7W37Q>
> .
> Triage notifications on the go with GitHub Mobile for iOS
> <https://apps.apple.com/app/apple-store/id1477376905?ct=notification-email&mt=8&pt=524675>
> or Android
> <https://play.google.com/store/apps/details?id=com.github.android&referrer=utm_campaign%3Dnotification-email%26utm_medium%3Demail%26utm_source%3Dgithub>.
>
>

Stx69 commented:
I tried for your current code add label to same line how checkbox and I'm weak to help you with this.. Looks to be connected with your current container style. I remmember need to be generate a extra "row" for appendchild. And to this "row" you put appenchild "label + checkbox".. But not sure.. If I will saw exmple i'll share here...

MJ659600 commented:
I figured out the checkbox.  Thanks for the help.

On Mon, Aug 30, 2021 at 5:25 PM Stx69 ***@***.***> wrote:

> I tried for your current code add label to same line how checkbox and I'm
> weak to help you with this.. Looks to be connected with your current
> container style. I remmember need to be generate a extra "row" for
> appendchild. And to this "row" you put appenchild "label + checkbox".. But
> not sure.. If I will saw exmple i'll share here...
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/darkforest-eth/plugins/pull/156#issuecomment-908188600>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ARO665DPQZTNRAEOIA7IA3DT7NFABANCNFSM5CU7W37Q>
> .
> Triage notifications on the go with GitHub Mobile for iOS
> <https://apps.apple.com/app/apple-store/id1477376905?ct=notification-email&mt=8&pt=524675>
> or Android
> <https://play.google.com/store/apps/details?id=com.github.android&referrer=utm_campaign%3Dnotification-email%26utm_medium%3Demail%26utm_source%3Dgithub>.
>
>

jacobrosenthal commented on content/artifacts/harvest-artifacts/plugin.js:
```diff
@@ -0,0 +1,351 @@
+        })
+    }
+
+    setInterval(() => {
```
This interval needs to be held so it can be cleared during the destroy phase. A bunch of plugins do this so you should be able to find an example :)

MJ659600 commented:
I just saw automover. Nicely done. Frankly, better code than mine.   Is
this PR still needed?

On Fri, Sep 3, 2021 at 11:37 PM Jacob Rosenthal ***@***.***>
wrote:

> ***@***.**** commented on this pull request.
> ------------------------------
>
> In content/artifacts/harvest-artifacts/plugin.js
> <https://github.com/darkforest-eth/plugins/pull/156#discussion_r701990831>
> :
>
> > +        this.minPlanetLevel,
> +        3,  //this is not used
> +        autoDeactivate
> +      )
> +        .then(errorPs => {
> +          failedPlanets.innerText = `${errorPs.length} needs intervention:`;
> +          for (let p of errorPs) {
> +            const pEntry = document.createElement('div');
> +            failedPlanets.appendChild(pEntry);
> +            pEntry.innerText = p;
> +            pEntry.onclick = () => { ui.centerLocationId(p) };
> +          }
> +        })
> +    }
> +
> +    setInterval(() => {
>
> This interval needs to be held so it can be cleared during the destroy
> phase. A bunch of plugins do this so you should be able to find an example
> :)
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/darkforest-eth/plugins/pull/156#pullrequestreview-746247212>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ARO665FKO3GRV4CL4DH5SXLUADTUPANCNFSM5CU7W37Q>
> .
> Triage notifications on the go with GitHub Mobile for iOS
> <https://apps.apple.com/app/apple-store/id1477376905?ct=notification-email&mt=8&pt=524675>
> or Android
> <https://play.google.com/store/apps/details?id=com.github.android&referrer=utm_campaign%3Dnotification-email%26utm_medium%3Demail%26utm_source%3Dgithub>.
>
>

jacobrosenthal commented:
Can always have multiple but yeah in this case maybe we can close out this one. 

Thanks for submitting, I know many people used this to clean out their artifacts the week after round end
